docs/Forkception.md

```diff
@@ -0,0 +1,75 @@
+# Forkception
+
+Fork is a special case within bionode-watermill because it requires to know 
+everything that follows the tasks within the `fork`. In current API there is
+ no easy way to get graph shape before actually running the `pipeline.js` 
+ scripts. Therefore, `fork` uses `join` to construct independent lineages for
+  each task within the `fork`.
+  
+  Currently `lib/orchestrator/fork.js` just collects the tasks inside `fork` 
+  and then everything else is handled by `lib/orchestrator/join.js`, i.e., 
+  the branching of the pipeline in several leaves. This branching requires 
+  that `fork` *downstream tasks* are multiplied as many times as the number 
+  of tasks inside `fork`.
+  
+  > For instance if fork contains 3 tasks the pipeline will have to branch
+  > 3 times from this point on:
+  >
+  >```javascript
+  >const pipeline = join(task0,fork(task1, task2, task3), task4)
+  >```
+  > This will end up in three different branches (or lineages) with three 
+  >leaves:
+  > `task0 --> task1 --> task4`, `task0 --> task2 --> task4` and `task0 --> 
```

Comment:
name these forked tasks `task2A, task2B, task2C`

(to be consistent with idea that fork is for multiple "alternatives" to a specific task (e.g. different params)
rather than different tasks

Reply:
:+1: